packages/x-date-pickers/src/DateTimePicker/DateTimePicker.types.ts

```diff
@@ -44,10 +44,11 @@ export interface DateTimePickerProps<
    */
   desktopModeMediaQuery?: string;
   /**
-   * Years rendered per row.
-   * @default 4 on desktop, 3 on mobile
+   * If `false`, display the years in their default order (chronological)
+   * If `true`, reverse the order of displayed years
+   * @default false
    */
-  yearsPerRow?: 3 | 4;
+  reverseYears?: boolean;
```

Comment:
This looks strange, did you really need explicit type here? 🤔 
Pickers should implicitly inherit props from `YearCalendar` props...
The only reason why `yearsPerRow` is redefined on the `Desktop` variant type is to have different defaults than for Mobile. 😉 

I think we should be able to avoid changes in these files, having it defined in `ExportedYearCalendarProps` should be enough. 🤞

Reply:
Right! I was having some issues with missing props and then I thought to just adjust those to be in alphabetical order but I will remove it here if not needed.

Reply:
I also agree that the picker works well with both the year and month views and I've come to learn that this feature could be useful even when the full date picker is in use. I'll add the month view!

I also notice an issue with the year only picker, that there will still be space reserved for the month view and so there's a blank space at the end of the years list. Adding the month view will prevent that from showing in the docs but I might work on it in a separate PR.

Reply:
> I also notice an issue with the year only picker, that there will still be space reserved for the month view and so there's a blank space at the end of the years list. Adding the month view will prevent that from showing in the docs but I might work on it in a separate PR.

Yes, that is a slight problem, however, it might not be trivial to solve, given that we rely on fixed width and height. 🤔 
If you fancy, feel free to check this issue out. 👍

Reply:
I will try to think about that once this first contribution is out of the way!

I did re-incorporate the other fixes to the test for the month to save hanging to do a whole PR for it.